File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/classes/nestedClass/staticNestedClass/OuterClass.java
package de.fraunhofer.iem.classesAndObjects.classes.nestedClass.staticNestedClass;

/**
 * Demonstrates the static nested class in Java
 *
 * @author Ranjith Krishnamurthy
 */
public class OuterClass {
    private int outerFlag = 110;

    public void printAll() {
        System.out.println(new InnerClass().innerFlag);
        System.out.println(outerFlag);
    }

    public static class InnerClass {
        private int innerFlag = 23421;

        public void printAll() {
            System.out.println(innerFlag);
        }
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/nestedAndInnerClass/innerClass/OuterClass.kt
package de.fraunhofer.iem.classesAndObjects.nestedAndInnerClass.innerClass

/**
 * Demonstrates the inner class in Kotlin
 *
 * @author Ranjith Krishnamurthy
 */
class OuterClass {
    private val outerFlag = 110
    fun printAll() {
        println(outerFlag)
    }

    inner class InnerClass {
        private val innerFlag = 23421
        fun printAll() {
            println(this.innerFlag)
            println(this@OuterClass.outerFlag)
        }
    }
}